rubix-bookkeeper/src/main/java/com/qubole/rubix/bookkeeper/BookKeeper.java

```diff
@@ -439,9 +440,11 @@ public void setAllCached(String remotePath, long fileLength, long lastModified,
     log.debug("Updating cache for " + remotePath + " StartBlock : " + startBlock + " EndBlock : " + endBlock);
 
     try {
-      md.setBlocksCached(startBlock, endBlock);
-      long currentFileSize = md.incrementCurrentFileSize((endBlock - startBlock) * CacheConfig.getBlockSize(conf));
-      replaceFileMetadata(remotePath, currentFileSize, conf);
+      OptionalInt updatedBlocks = md.setBlocksCached(startBlock, endBlock);
```

Comment:
Should we make the data type of startBlock and endBlock `int` ? Eventually when the block no. is set in bitmap then it is cast to int anyway. Can do this change in a different PR.

Reply:
Yes, that can be done. Block number in int can represent the file size of 2PB, we dont expect single file to be that big.

Reply:
Opened https://github.com/qubole/rubix/issues/369